purchase_work_acceptance/models/purchase.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2019 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models, _
+
+
+class PurchaseOrder(models.Model):
+    _inherit = 'purchase.order'
+
+    wa_count = fields.Integer(
+        compute='_compute_wa_ids',
+        string='WA count',
+        default=0,
+    )
+    wa_ids = fields.Many2many(
```

Comment:
Maybe you can use a One2many as you already have purchase_id on the WA.

Reply:
Edited : https://github.com/OCA/purchase-workflow/pull/774/commits/74b8c5f27338d714143dfee5372105bf91e269cc